src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -100,6 +104,21 @@ const RecordButton = (props: RecordButtonProps) => {
 };
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
+  function serialButtonClasses(){
+    let status;
+    const { serialDevice } = props;
+    const physicalConnect = localStorage.getItem("last-connect-message") === "connect";
+    const physicalDisconnect = localStorage.getItem("last-connect-message") === "disconnect";
+    const nodesInNeed = serialDevice.serialNodesCount > 0;
+    const hasPort = serialDevice.hasPort();
+
+    status = hasPort ? "has-port" : "no-port";
+    status += physicalConnect ? " physical-connection" : " no-physical-connection";
```

Comment:
Not necessary, but see `clsx` for creating class strings.

Reply:
I agree with the sentiment that a utility should be used rather than string manipulation, but while `clsx` is used in other projects, the utility used for that purpose in CLUE is `classnames`.

Reply:
refactored to use classnames - was [much shorter](https://github.com/concord-consortium/collaborative-learning/commit/f88c700f4b8a0e8d86edfdb7c38fa2826d71cc80):